crimson/star_fusion.py

```diff
@@ -41,70 +109,209 @@
-        "geneName": lrgname,
-        "geneID": lrgid,
-        "chromosome": lrchrom,
-        "position": int(lrpos),
-        "strand": lrstrand,
-    }
+    if break_side == "left":
+        gene = entries["LeftGene"]
+        breakpoint = entries["LeftBreakpoint"]
+        prefix = "Left"
+    elif break_side == "right":
+        gene = entries["RightGene"]
+        breakpoint = entries["RightBreakpoint"]
+        prefix = "Right"
+    else:
+        raise RuntimeError("Please specify either right or left")
+    gname, gid = gene.split(_DELIM["gids"])
+    chrom, pos, strand = breakpoint.split(_DELIM["loc"])
+
+    breakpoint_side = {
+        "geneName": gname,
+        "geneID": gid,
+        "chromosome": chrom,
+        "position": int(pos),
+        "strand": strand,
+    }  # type: Dict[str, Union[str, int]]
+
+    # Get the other side-specific fields from the output
+    sided_fields = [field for field in entries if field.startswith(prefix)]
+
+    # Remove the ones we already parsed
+    if prefix == "Left":
+        sided_fields.remove("LeftGene")
+        sided_fields.remove("LeftBreakpoint")
+    else:  # prefix is either "Left or "Right":
+        sided_fields.remove("RightGene")
+        sided_fields.remove("RightBreakpoint")
+
+    for field in sided_fields:
+        # Remove side from field name
+        new_field = field[len(prefix):]
+        # Convert to camelCase
+        camel_case = new_field[0].lower() + new_field[1:]
+        breakpoint_side[camel_case] = entries[field]
+
+    return breakpoint_side
+
+
+def parse_read_columns(
+    colnames: List[str],
+    values: List[str]
+) -> Tuple[Dict[str, str], Dict[str, List[str]]]:
+    """Parse the read columns out and return them seperately
+
+    The JunctionReads and SpanningFrags columns can contain the actual reads,
+    or just the counts of the reads. This makes parsing them correctly quite
+    convoluted.
+
+    Assumption: The last occurrence of the JunctionReads and SpanningFrags
+    columns in the file contain the actual reads, while the first one contains
+    the counts.
+    If there is only one (as is the case for v1.6.0), it contains the reads
+
+    :param values: List of values from the file
+    :para colnames: List of column names from the file
+    """
+    entries = dict()
+    reads = dict()
+
+    # Keep the column names and values together
+    together = [(name, val) for name, val in zip(colnames, values)]
+
+    # Extract the JunctionReads
+    rev = together[::-1]
+    for name, val in rev:  # pragma: no branch
+        if name == "JunctionReads":
+            together.remove((name, val))
+            break
+    reads[name] = val.split(",")
+
+    # Extract the SpanningFrags
+    for name, val in rev:  # pragma: no branch
+        if name == "SpanningFrags":
+            together.remove((name, val))
+            break
+    reads[name] = val.split(",")
+
+    # The other columns are regular entries
+    entries = {k: v for k, v in together}
```

Comment:
Instead of iterating over `rev` and then doing `list.remove((name, val))`, what do you think about directly iterating over `together` (or `zip(colnames, values)`) and then constructing the final dictionary?

The `JunctionReads` / `SpanningFrags` issue can be tricky, but how about simply using `str.isdigit` to check if the raw value represents a count or not?

So somewhere inside the loop, we can a check that looks like:
```
...
    if name in {"JunctionReads", "SpanningFrags"}:
        if value.isdigit():
            entries[f"n{name}"] = int(value)
        else:
            entries[name] = value.split(",")
    else:
        entries[name] = value
...
```

P.S. I have to say, I was happy reading the docstring for this function by the way, it made the code that much easier to understand :+1: